raven/src/main/java/com/getsentry/raven/RavenContext.java

```diff
@@ -0,0 +1,117 @@
+package com.getsentry.raven;
+
+import com.getsentry.raven.event.Breadcrumb;
+import com.getsentry.raven.util.CircularFifoQueue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * RavenContext is used to hold {@link ThreadLocal} context data (such as
+ * {@link Breadcrumb}s) so that data may be collected in different parts
+ * of an application and then sent together when (e.g.) an exception occurs.
+ */
+public class RavenContext implements AutoCloseable {
+
+    /**
+     * Thread local set of activate context objects. Note that an {@link IdentityHashMap}
+     * is used instead of a Set because there is no identity-set in the Java
+     * standard library.
+     */
+    private static ThreadLocal<IdentityHashMap<RavenContext, RavenContext>> activeContexts =
```

Comment:
I'm not totally clear on what this is for — is this supposed to facilitate passing a `RavenContext` instance between different threads? If this is the case, it might be a good idea to add something to the `Breadcrumb` object that lets us record the current thread ID and/or name. As far as I know there isn't anything in the "specification" for this yet, but it seems like a thing we'd want to add for platforms where concurrency is going to be more prevalent.